### DISCUSSION TITLE
My qtbase installation gets larget, along with others. Is this a problem?

### DISCUSSION THREAD

**Discussion by KRJackLee** [microsoft/vcpkg#32347] [Q&A]
I am using VS 2022 MSVC + CMake for C++ program on Windows, and using vcpkg for package management. I recently tried to install qt, or qtbase, as an attempt to apply UI to my program. However, the installation gets large, qtbase requires tens of dependencies and cost around 30 or more gigabytes in total during installation and build. The syntax is `vcpkg install qtbase:x64-windows`. Is it a system-wide issue specific to my env setting or something, or is it just me who is experiencing this issue? My installation of boost also costs tens of gigabytes, syntax `vcpkg install boost:x64-windows`. Or is it just that Windows isn't the first choice for Qt+cpp? 
By the way, if I encounter an installation that costs too much space for me wanting to revert the operation, is there any vcpkg command doable for reverting the package environment back to previous state? Manual delete is tedious, let alone there may be dependencies of other packages is updated during the installation of this costly syntax. Thx.

**Comment by autoantwort** [comment]
I think this is normal :)

**Comment by autoantwort** [comment]
You could pass `--clean-buildtrees-after-build --clean-packages-after-build` to vcpkg when building packaged to make your installation smaller. You can also always delete the `packages` and `buildtrees` folders.

**Comment by Osyotr** [comment]
`vcpkg install qtbase:x64-windows` this also installs default features which may drag more dependencies. Use `core` feature and add everything you need `vcpkg install qtbase[core,widgets,network]:x64-windows`.
For boost, it's preffered to use individual ports like `boost-asio` for the very same reason.
